plugin.xml

```diff
@@ -78,11 +78,8 @@ xmlns:android="http://schemas.android.com/apk/res/android"
 
         <config-file target="AndroidManifest.xml" parent="/*">
             <uses-permission android:name="android.permission.RECORD_AUDIO" />
-            <uses-permission android:name="android.permission.READ_MEDIA_AUDIO" />
-            <uses-permission android:name="android.permission.READ_MEDIA_IMAGES" />
-            <uses-permission android:name="android.permission.READ_MEDIA_VIDEO" />
-            <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" android:maxSdkVersion="32" />
-            <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" android:maxSdkVersion="32" />
+            <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" android:maxSdkVersion="29" />
+            <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" android:maxSdkVersion="29" />
```

Comment:
No need to change the `android:maxSdkVersion="32"` nor the check on `TIRAMISU` for `storagePermissions`.

As said and set in PR #262, it's `In API 33 these two permissions are ineffective` that explains the `32`, required to prevent breaking the build, and in between 29 and 32 yes it may depend or be useless but even if declared here or requested by the app, Android deals with it seamlessly for us.
And "all" other plugins with such `<uses-permission` on `EXTERNAL_STORAGE` and check on `TIRAMISU` did the same, not less: [cordova-plugin-camera](https://github.com/apache/cordova-plugin-camera/blob/master/plugin.xml#L58), [cordova-diagnostic-plugin](https://github.com/dpa99c/cordova-diagnostic-plugin/blob/master/src/android/Diagnostic_Camera.java#L60) ; while [cordova-plugin-file](https://github.com/apache/cordova-plugin-file/blob/master/README.md#sdk-target-less-than-29) let you decide and do it, but with an even stricter `28`.

Reply:
For some context, `WRITE_EXTERNAL_STORAGE` permission didn't grant any more privileges over `READ_EXTERNAL_STORAGE` since API 29/scoped storage. But requesting the write permission still implicitly granted the `READ_EXTERNAL_STORAGE` permission which had meaning until API 33.

There might still be Cordova code relying on that behavour and that's why API 32 was chosen for the max SDK version. It probably shouldn't be changed, especially `READ_EXTERNAL_STORAGE`

I think in order to appease Google, we will have to do what we did for the file plugin, which is to remove all permissions and have them opt in if the app requires them. But that isn't a complete solution alone...

I think the main problem is this plugin relies on using the `File` APIs to read the content, which isn't part of the MediaStore scoped storage. That's why Cordova depends on the `READ_MEDIA_*` permissions right now. I'm pretty sure a proper native solution would use the `content://` urls provided by the MediaStore APIs which is a special link to the file within the media containers with temporary read (and potentially write) grants to that specific media resource. So I'm pretty sure Cordova has to start sharing the `content://` urls but I'm I think `content://` urls have limited support for reading (or writing).

We have experimented doing a file copy to the temp/cache folder in the past which works well for images and audio since they are going to be rather small files, so the copy operation will be pretty quick, and then you can access the file from the cache copy with standard file APIs without additional permissions. This strategy doesn't work very well for videos since they tend to be much larger.

Reply:
Maybe I don't have the full picture here, but I'm pretty sure we don't need this permission for Android 29+ for **medias** as stated in the following documentation : https://developer.android.com/training/data-storage/shared/media?hl=en#storage-permission and regarding the tests that I have performed.
Whats bugs me is that it add a permission popup on the first media capture intent.

Anyway, do you want me to rollback `android:maxSdkVersion="32"` ?

Reply:
`rollback to android:maxSdkVersion="32"?` to me yes, and the check on `TIRAMISU` for `storagePermissions` too

About the "broad" permissions, I would warn about the default Android/Google apps behavior that mostly writes and return the captured files following like the best practices, and so with the official emulators and any Pixel device we often have no issue. While apps from Google Play and other manufacturers make things more difficult, writing in there own directories, not shared or not as "default" expected.
The point on "[files that your app owns](https://developer.android.com/training/data-storage/shared/media?hl=en#storage-permission)" is not always easy in this plugin scenario, delegating capture through an intent, and so which app really owns the file? I always considered we need these permissions for this case and so that your app could access the file written by the capture app on intent return (because owned by the capture app and not yours?).

Reply:
Understood, I force-pushed to remove the last commit that changed the `maxSdkVersion` for storage permissions instead of doing an unescessary revert commit.

I re-tested on Android 9, 10, 11, 12L, 13 and 14, everything is still fine.

Reply:
@bmarsaud did you test both audio and video recording? I will do it myself soon, however, I do not have an Android 12 device anymore.